pkg/subnetmanager/subnet_informer.go

```diff
@@ -392,8 +398,11 @@ func (sc *SubnetController) syncControlledIPPoolIPs(ctx context.Context, subnet
 			}
 
 			if !exist {
-				logger.Sugar().Infof("The Application %v corresponding to the auto-created IPPool %s no longer exists, remove the pre-allocation from Subnet", appNamespacedName, poolName)
-				continue
+				_, err := sc.IPPoolsLister.Get(poolName)
+				if apierrors.IsNotFound(err) {
+					logger.Sugar().Infof("The Application %v corresponding to the auto-created IPPool %s no longer exists, remove the pre-allocation from Subnet", appNamespacedName, poolName)
+					continue
+				}
 			}
 
 			autoIPPoolIPs, err := spiderpoolip.ParseIPRanges(*subnet.Spec.IPVersion, preAllocation.IPs)
```

Comment:
Based on the provided code patch, I can make the following observations:

1. In the `UpdateFunc` of the `ipPoolInformer`, the new and old state of the object is compared using `reflect.DeepEqual`. While this approach works for simple comparisons, it may not always produce correct results in more complex scenarios. A better approach would be to use a library such as `github.com/google/go-cmp/cmp` which provides better handling of nested structures and types.

2. In the `syncControlledIPPoolIPs` function, there is a check for whether the corresponding application still exists before removing pre-allocated IPs from the subnet. However, the check itself can fail if the application was deleted concurrently. A better approach would be to catch possible errors when retrieving the app and log them accordingly.

3. It's difficult to perform a comprehensive code review without seeing the entire file or package that this code belongs to. It's important to ensure that the code follows best practices, is easy to read and understand, and has appropriate error handling throughout.

4. Lastly, commenting the code properly can help future maintainers to understand the logic behind the code better.